### PR TITLE
add pre-commit hooks for code formatting

### PR DESCRIPTION
**Pull request by mayank31398**
Adding pre-commit hooks for code formatting to this repo.
For now I have kept your code excluded.
Let me know if you don't mind me including all the python files
@stas00

### CODE REVIEW

File: bloom-inference-server/benchmark.py [hunk 1]
```diff
@@ -5,12 +5,18 @@
...
+from models import get_model_class
+from utils import BENCHMARK
+from utils import DS_INFERENCE
+from utils import DS_ZERO
+from utils import GenerateRequest
+from utils import get_argument_parser
+from utils import get_dummy_batch
+from utils import parse_generate_kwargs
+from utils import print_rank_n
+from utils import run_and_log_time
```
Comment (stas00) [inline bloom-inference-server/benchmark.py:19]: this looks quite bad.

perhaps we adopt the same setup as `transformers`? If it resonates here is the setup to copy:

https://github.com/huggingface/transformers/blob/0d1ba2dd0ba1b584c0380ee9f69c2b222b462d76/Makefile#L68-L69

https://github.com/huggingface/transformers/blob/0d1ba2dd0ba1b584c0380ee9f69c2b222b462d76/setup.cfg#L1-L47

https://github.com/huggingface/transformers/blob/0d1ba2dd0ba1b584c0380ee9f69c2b222b462d76/pyproject.toml#L1-L3

Comment (stas00) [reply bloom-inference-server/benchmark.py:19]: then yes, by all means you can expand it to the rest of the code.

**Comment by mayank31398** [comment]
Done @stas00 let me know if it looks good now

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    tensorboardX
+    tensorflow
+    tensorflow_datasets
+    timeout_decorator
+    torch
+    torchaudio
+    torchtext
+    torchvision
+    torch_xla
+    tqdm
```
[the remark this answers is not in the document]
Comment (mayank31398) [reply setup.cfg:42]: Yes, this is for grouping imports.
I feel like we can leave it as is.
I doubt we will even need torchvision/audio ... but seems aligned with transformers repo.
I don't have any specific packages in mind, I will add if needed in the future.

[the remark this answers is not in the document]
Comment (stas00) [reply setup.cfg:42]: that works.

**Pull request merged**